### PR TITLE
batch rotate key upak loads

### PR DESCRIPTION
**Pull request by heronhaye**
I tried to simplify code and leave existing logic unchanged. 

There is also a function loadGroups that could benefit from batching as well. Perhaps we can change it over in a future ticket.

### CODE REVIEW

File: go/libkb/upak_loader.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ type UPAKLoader interface {
 	PutUserToCache(ctx context.Context, user *User) error
 	LoadV2WithKID(ctx context.Context, uid keybase1.UID, kid keybase1.KID) (*keybase1.UserPlusKeysV2AllIncarnations, error)
 	CheckDeviceForUIDAndUsername(ctx context.Context, uid keybase1.UID, did keybase1.DeviceID, n NormalizedUsername, suppressNetworkErrors bool) error
-	Batcher(ctx context.Context, getArg func(int) *LoadUserArg, processResult func(int, *keybase1.UserPlusKeysV2AllIncarnations) error, window int) (err error)
+	Batcher(ctx context.Context, getArg func(int) *LoadUserArg, processResult func(int, *keybase1.UserPlusKeysV2AllIncarnations) error, ignoreError func(err error) bool, window int) (err error)
```
Comment (heronhaye) [inline go/libkb/upak_loader.go]: If you think it's better I can make this take an options struct or split it out into different functions so callers aren't passing in default values.

Comment (jakob223) [reply go/libkb/upak_loader.go]: Eh it's probably fine.. 🤷

**Comment by heronhaye** [comment]
I did some more digging and found out that with WithPublicKeyOptional, LoadUser will never return a NoKeyError/AccountResetError. So I was able to remove the ignoreError func.

**Comment by heronhaye** [comment]
deleted old comment and merging over previous flakes

**Pull request merged**